src/ACEtk/interpretation/ContinuousEnergyNeutron/Builder/EnergyDistribution/Law2Int.hpp

```diff
@@ -0,0 +1,16 @@
+struct Law2Int{
+  int operator()( const TabularEquiprobableEnergyBins&   ) const { return 1; }
+  int operator()( const DiscretePhotonEnergy&            ) const { return 2; }
+  int operator()( const LevelScattering&                 ) const { return 3; }
+  int operator()( const ContinuousTabularDistribution&   ) const { return 4; }
+  int operator()( const GeneralEvaporationSpectrum&      ) const { return 5; }
+  int operator()( const SimpleMaxwellianFissionSpectrum& ) const { return 7; }
+  int operator()( const EvaporationSpectrum&             ) const { return 9; }
+  int operator()( const EnergyDependentWattSpectrum&     ) const { return 11; }
+  int operator()( const TabularLinearFunctions&          ) const { return 22; }
+  int operator()( const TabularEnergyMultipliers&        ) const { return 24; }
+  int operator()( const KalbachMann&                     ) const { return 44; }
+  int operator()( const KalbachMannAngularDistribution&  ) const { return 61; }
+  int operator()( const NBodyPhaseSpace&                 ) const { return 66; }
+  int operator()( const LaboratoryAngleEnergy&           ) const { return 67; }
```

Comment:
Maybe do these as constexpr? And they could be declared on the actual objects as well: DiscretePhotonEnergy nows that it is a LAW=1. By visiting on the variant you can then derive the LAW without having to go through this. ENDFtk uses this construct/pattern a lot.

Reply:
We could make this a function of each individual object, but it wouldn't remove the need for `Law2Int`. We use a `variant` of the different energy distribution laws. We would still have to visit on the variant which is what this struct does.